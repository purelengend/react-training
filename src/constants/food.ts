import { Food } from '@components/common/Cards/ProductCard';

export const FOOD_NAME_WARNING_MSG =
  'The food name cannot be empty or have two trailing spaces consecutive.';
export const FOOD_PRICE_WARNING_MSG =
  'The food price cannot be empty and greater than 0';
export const FOOD_IMG_WARNING_MSG =
  'The food image URL cannot be empty and must be in the right format';
export const FOOD_QUANTITY_WARNING_MSG =
  'The food quantity cannot be empty and must be an integer that greater than 0';
export const EMPTY_MSG =
  'Oops! It looks like there are no food items available at the moment.';
export const DEFAULT_FOOD_ID_VALUE = '';

export const defaultData: Food = {
  id: '',
  name: '',
  price: 0,
  imageUrl: '',
  quantity: 0,
  createdAt: new Date()
};

export interface FoodErrorMessage {
  name: string;
  price: string;
  imageUrl: string;
  quantity: string;
}

export const defaultFoodErrorMessage: FoodErrorMessage = {
  name: '',
  price: '',
  imageUrl: '',
  quantity: ''
};
