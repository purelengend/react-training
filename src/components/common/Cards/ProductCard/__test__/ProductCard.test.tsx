import { defaultData } from '@constants/food';
import { customRenderer, render } from '@src/test/test-utils';
import userEvent from '@testing-library/user-event';

import { ProductCard } from '..';

const mockFood = defaultData;

describe('ProductCard test cases', () => {
  it('should render correctly with required functions props', () => {
    const mockOnClick = vi.fn();

    const mockProductCard = customRenderer(
      <ProductCard
        product={mockFood}
        onDeleteClick={mockOnClick}
        onEditClick={mockOnClick}
      />
    );

    expect(mockProductCard.toJSON()).toMatchSnapshot();
  });

  it('should invoke functions when being clicked', async () => {
    const mockOnDeleteClick = vi.fn();
    const mockOnEditClick = vi.fn();

    const { container } = render(
      <ProductCard
        product={mockFood}
        onDeleteClick={mockOnDeleteClick}
        onEditClick={mockOnEditClick}
      />
    );

    const mockButtonList = container.getElementsByTagName('button');

    await userEvent.click(mockButtonList[0]);

    expect(mockOnDeleteClick).toBeCalled();

    await userEvent.click(mockButtonList[1]);

    expect(mockOnEditClick).toBeCalled();
  });
});
