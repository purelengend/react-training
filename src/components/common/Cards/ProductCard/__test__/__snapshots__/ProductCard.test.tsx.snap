// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductCard test cases > should render correctly with required functions props 1`] = `
<div
  className="d-flex-center d-flex-col _product-card_6cffd1"
>
  <button
    className="d-flex-center _secondary-icon_6cffd1 _delete-btn_6cffd1"
    data-id="0"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <img
      alt="Cross Icon"
      data-id=""
      height="20"
      src="/src/assets/icons/cross-icon.svg"
      width="20"
    />
  </button>
  <div
    className="d-flex-col _product-wrapper_6cffd1"
  >
    <img
      alt=""
      className="_primary-product_6cffd1"
      height="127"
      src=""
      width="127"
    />
    <div
      className="d-flex-col _product-content-wrapper_6cffd1"
    >
      <p
        className="_product-name_6cffd1"
      />
      <div
        className="d-flex-center _product-detail_6cffd1"
      >
        $ 
        0
        <div
          className="_separate_6cffd1"
        />
        0
         Bowls
      </div>
    </div>
  </div>
  <button
    className="d-flex-center _product-mutation_6cffd1 undefined"
    data-id="0"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <img
      alt="Edit Icon"
      className="primary-icon undefined"
      src="/src/assets/icons/edit-icon.svg"
    />
    <p
      className="_mutation-content_6cffd1 undefined"
    >
      Edit dish
    </p>
  </button>
</div>
`;
